### PR TITLE
Build with CMake, include GoogleTest and first test

### LINKED ISSUE #3: Replace current Makefile with CMake
We want to replace the current Makefile by a build routine that can be executed with CMake.

### PR DESCRIPTION
**Pull request by mknaranja**
Closes #3 #4 #5 #6

### CODE REVIEW

File: tests/comparetest.cpp [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+{
+    gmgpolar gmgtest;
+    gmgtest.create_grid_polar(); //only the finest grid is now created
+    int finest_nodes = gmgtest.v_level[0]->nr * gmgtest.v_level[0]->ntheta;
+    EXPECT_EQ(gmgtest.v_level.size(), 1);
+    //std::cout<<std::stoi(input[initparam+1])<<std::endl ;
+    EXPECT_EQ(finest_nodes, std::stoi(input[initparam + 1]));
+    //EXPECT_EQ(gmgtest.v_level[0]->nr, std::stoi(input[initparam + 2])); //Segmentation fault ???
+}
+TEST_F(Test_Parameters, Multigrid)
```
Comment (mknaranja) [inline tests/comparetest.cpp:67]: Please also provide better test names.

File: thirdparty/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+message(STATUS "Adding GoogleTest")
+
+
+include(FetchContent)
+FetchContent_Declare(
+  googletest
+  URL https://github.com/google/googletest/archive/03597a01ee50ed33e9dfd640b249b4be3799d395.zip
```
Comment (mknaranja) [inline thirdparty/CMakeLists.txt:7]: Do we need a specific zip from the archive?

File: include/writetofile.h [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#include <fstream>
+
+namespace ouin
```
[the remark this answers is not in the document]
Comment (CodingAllan) [reply include/writetofile.h:3]: I didnt copy it. I just wanted to make sure that the name of the function is unique. But it isnt neccesary

File: tests/comparetest.cpp [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+#include "cmdline.h"
+#include "gmgpolar.h"
+
+class Test_Parameters : public ::testing::Test
+{
+protected:
+    void SetUp() override
+    {
+        //Results of the run with default parameters. 
+        input.assign({5,0,0,4,4,3,0,1,0,3136,49,64,4,12}) ;
```
Comment (mknaranja) [inline tests/comparetest.cpp:14]: You don't have correct formatting. Please check for all files and commit potential adjustments.

File: thirdparty/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,14 @@
+message(STATUS "Adding GoogleTest")
+
+
+include(FetchContent)
+FetchContent_Declare(
+  googletest
+  GIT_REPOSITORY https://github.com/google/googletest.git
+  GIT_TAG release-1.12.1
```
Comment (mknaranja) [inline thirdparty/CMakeLists.txt:8]: Why not v1.13.0?

**Review by CodingAllan — APPROVED** [review]

**Pull request merged**